### PR TITLE
Using time arrays in PinchedModel.get_initial_spectra

### LINKED ISSUE #221: Allow array as `t` argument for `SupernovaModel.get_*_spectra`
# What I want

As a user I need to build the neutrino spectrum vs. time. I want to be able to calculate this 2d spectra directly from `SupernovaModel` without constructing loops on the `t` variable:

```python
#Loading the SN model
model = ccsn.Bollig_2016('models/Bollig_2016/s11.2c')
#time and energy bins
t = np.linspace(0,10,101)<<u.s
E = np.linspace(0,50,51)<<u.MeV

fluxes2d = model.get_initial_spectra(t,E)
assert fluxes2d[0].shape == (101, 51)

#backward compatibility: if we provide a scalar time we get 1D arrays.
#Let's take t[20] as scalar value
fluxes1d = model.get_initial_spectra(t[20],E)
assert fluxes1d[0].shape == (51)

#check that 1d and 2d results are compatible
for flavor in fluxes2d:
   assert np.allaclose(fluxes2d[flavor][20], fluxes1d)
```
# What we have now
Currently `SupernovaModel.get_initial_spectra` and `SupernovaModel.get_transformed_spectra` can work only with a single `t` value. As far as I understand, the only limitation is in the implementration of `PinchedModel`, so it can be easily corrected.

This is in addition to the existing `test_snowglobes.py` tests, which should work without modification and provide crosscheck with the events table.
# Outlook
This should make possible further optimization in `snowglobes.generate_*` functions - getting rid of the time loops there can speed up computation and make code cleaner.

After implementing #214 we can use `NeutrinoFlux` class as the return value.

### PR DESCRIPTION
**Pull request by Sheshuk**
Closes #221

### CODE REVIEW

File: python/snewpy/models/base.py [hunk 1]
```diff
@@ -238,6 +238,9 @@ def get_initial_spectra(self, t, E, flavors=Flavor):
         initialspectra : dict
             Dictionary of model spectra, keyed by neutrino flavor.
         """
+        t = u.Quantity(t, ndmin=1)
+        E = u.Quantity(E, ndmin=2)
+
```
Comment (mcolomermolla) [inline python/snewpy/models/base.py:243]: I don't understand why both arrays don't have ndmin = 1 but for energy it is 2.

Comment (Sheshuk) [reply python/snewpy/models/base.py:243]: You're right, it's not very clear here. 
1) I wanted to ensure that both `t` and `E` are converted to arrays even if they are only scalars
2) I wanted them to be 2D arrays, so everything relevant to `E` would be of shape `(len(E),1)` and `t` would be `(1, len(t))` - so after the calculations we'll have 2D array of shape `(len(E), len(t))`. **But** `t` will be used later in the `np.interp` method, which works only for 1D arrays. So I had to keep it 1D, and produce 1D arrays `L`,`Ea`,`a`, and later turn them to shape `[1, len(t)]` using `np.expand_dims`.

I will update the code to make it more clear.

Comment (Sheshuk) [reply python/snewpy/models/base.py:246]: Please have a look now

Comment (mcolomermolla) [reply python/snewpy/models/base.py:246]: Okey, thanks for clarifying, I get now what you did, it makes sense to me

**Comment by JostMigenda** [comment]
Minor note: If `t` can now also be an array, the type of `t` in the docstring should be updated accordingly.

**Comment by Sheshuk** [comment]
> Minor note: If `t` can now also be an array, the type of `t` in the docstring should be updated accordingly.

`astropy.Quantity` is an extension of ndarray, so we're actually using them, not `ndarray of astropy.Quantity`, for example
```python
>>> T = np.linspace(0,10) << u.s
>>> type(T)
astropy.units.quantity.Quantity
```
so the current docstring is (now even more) correct

**Review by JostMigenda — APPROVED** [review]
Good point about the types! It’s a bit unfortunate that this distinction cannot be expressed right now, but maybe that’ll change in the coming years as a result of [PEP 646](https://peps.python.org/pep-0646/).

This PR LGTM; thanks, Andrey!

**Comment by Sheshuk** [comment]
Thanks! Yes these variadic generics look like they would make the signatures much cleaner.
In general I think it would be good to add the type hints throughout snewpy, but that's a separate task.

**Pull request merged**